### DISCUSSION TITLE
Ursina slowed when using threading instead of "from direct.stdpy import thread"

### DISCUSSION THREAD

**Discussion by kowshik1234** [pokepetter/ursina#616] [Q&A]
When using the threading library instead of the `from direct.stdpy import thread` I am seeing insane lag and performance reduction. I am not sure what is special in the direct.stdpy and why is it reducing the performance. My app details are provided below

1. Loads a calibration window playing this animation for the [loading screen example] (https://github.com/pokepetter/ursina/blob/10a5ea89fcfcc3d5e231a1ce5f95848d1963bcfa/samples/loading_screen.py) 
2. During the calibration window, the objects are loaded for the first time and cached so the next iteration will be easier to load. I have used a thread to load these models in the background when the calibration window runs. 
3. However, the calibration animation runs very very slow when the models are being loaded when threading library is used instead of direct.stdpy's thread.
4. But I am using the threading to read COM ports and display the data. Those data queues are working at high speed and my program is able to read them,.

On the contrary,

If I use the `from direct.stdpy import thread` for my application, then the animations are loading smooth but the COM port data which is being read is getting choked and the enqueuing and dequeuing speeds are very very poor. 

Please suggest how can I improve my speed for the animation without losing the threading for the COM port. Let me know if I confused you.

Thanks

**Comment by pokepetter** [comment]
Why can't you use thread from direct.stdpy? It's a direct replacement for python's threading module, except it's supported.
See: https://docs.panda3d.org/1.10/python/programming/tasks-and-events/threading

**Reply by kowshik1234** [reply]
I tried using the direct.stdpy, as I told when i use it, the COM port data that I am receiving is choking the data Queue. It's dequeuing very slow.
